app/components/Home/WalletLauncher.js

```diff
@@ -132,19 +155,65 @@ class WalletLauncher extends React.Component {
     }
   }
 
-  onSubmit = values => {
-    const { startLnd } = this.props
+  onSubmit = async values => {
+    const { startLnd, wallet } = this.props
+    //for the remote wallets we need to initiate lndconnect URI re-generation before launching
+    if (wallet.type !== 'local' && this.hasChanges()) {
+      const config = await this.saveSettings()
+      return startLnd(config)
+    }
     return startLnd(formToWalletFormat(values))
   }
 
-  saveSettings = () => {
-    const { putWallet, showNotification, showError, intl } = this.props
+  /**
+   * Saves current lnd config based on current form state
+   * Returns the actual config that was saved
+   * @memberof WalletLauncher
+   */
+  saveSettings = async () => {
+    const { putWallet, showNotification, showError, intl, wallet } = this.props
     try {
-      const formState = this.formApi.getState()
+      const { formApi } = this
+      const formState = formApi.getState()
       const { values } = formState
-      putWallet(formToWalletFormat(values))
+      let result = values
+
+      // for the remote wallets we need to re-generate lndconnectUri and QR using updated
+      // host, cert and macaroon values. This is done in the main process
+      // this process is skipped if original lndconnect uri contains raw cert or macaroon and not paths
+      if (wallet.type !== 'local' && !isEmbeddedLndConnectURI(wallet.lndconnectUri)) {
+        ipcRenderer.send(
```

Comment:
Should we be using `dispatch` here rather than sending via `ipcRenderer` directly? I think there is only one other place in the codebase where we do this, and I'm not particularly happy about that. All other IPC messages are sent using `send` from `redux-electron-ipc`

Reply:
By dispatch you mean `redux dispatch` ? If so, then this doesn't update the redux state so doesn't seem appropriate here

Reply:
It does in the sense that It logs the action in the redux history so that you have visibility into the fact that an IPC message was sent within the redux history. This is how all of our IPC messages are sent.

Reply:
I get that it's how it's currently done in most places, but benefits are not clear to me. Especially when seeing code like

Reply:
I would definitely agree with that. The whole process of using IPC is horrible IMO. Sending off events and hoping that certain other events come back in response... I'd much rather see a fully promise based approach to communicating with the main process. But short of that sort of wider change, I do feel there is some merit to taking a consistent approach across the board.

Reply:
Updated